pkgs/development/libraries/mpich/default.nix

```diff
@@ -1,24 +1,50 @@
-{ lib, stdenv, fetchurl, perl, gfortran
-,  openssh, hwloc
+{ stdenv, lib, fetchurl, perl, gfortran
+,  openssh, hwloc, autoconf, automake, libtool
+# device options are ch3 or ch4
+, device ? "ch4"
+# backend option are libfabric or ucx
```

Comment:
Please add an assert to check if it is either libfabric or ucx.

Reply:
done